client/src/components/Admin/CourseUpload/CourseUpdate.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useRef } from "react";
+import JoditEditor from "jodit-react";
+import "./CourseUpload.css";
+import FormQuiz from "./FormQuiz/FormQuiz";
+import { useParams } from "react-router";
+import { useNavigate } from "react-router";
+import GetCourse from "../../../data/GetCourse";
+
+export default function CourseUpdate() {
+  const params = useParams();
+  const navigate = useNavigate();
+
+  const oldData = GetCourse(`/chapters/${params.id}`);
+  const editor = useRef(null);
+  const [formData, setFormData] = useState();
+  if (oldData && !formData) setFormData(oldData);
+
+  const updateChapter = async (e) => {
+    e.preventDefault();
+    const { category, chapter, content, quiz, topic, chapterId } = formData;
+    try {
+      const res = await fetch("/update", {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          id: formData._id,
+          category,
+          chapter,
+          content,
+          quiz,
+          topic,
+          chapterId,
+        }),
+      });
+      const data = await res.json();
+      navigate(`/learning/${chapter}`);
+      if (data.status === 422 || data.status === 500) {
+        alert(data.error);
+        return data.error;
+      } else {
+        alert(data.message);
+      }
+    } catch (err) {
+      console.log("An erros occured: " + err);
+    }
+  };
+
+  function onChangeHandler(e) {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  }
+
+  return (
+    <div className="course-upload-container">
+      <h1 className="heading">UPDATE the course</h1>
+      {formData && (
+        <form method="PUT" onSubmit={updateChapter} className="course-form">
+          <div className="form-item-wrapper">
+            <div className="form-item">
+              <label htmlFor="course-category">Category</label>
+              <select
+                name="category"
+                id="course-category"
+                onChange={onChangeHandler}
+                defaultValue={formData.category}
+              >
+                <option value={"Basics"}>Basics</option>
+                <option value={"DSA Starter"}>DSA Starter</option>
+                <option value={"Adv. DSA"}>Adv. DSA</option>
+              </select>
+            </div>
+            <div className="form-item">
+              <label htmlFor="course-topic">Topic</label>
+              <select
+                name="topic"
+                id="course-topic"
+                onChange={onChangeHandler}
+                defaultValue={formData.topic}
+              >
+                <option value={"Basics of C++"}>Basics of C++</option>
```

Comment:
Will recommend to use map in similar scenarios and store all the topics in array.